lib/src/SaiInterface.cpp

```diff
@@ -173,3 +173,275 @@ sai_status_t SaiInterface::get(
+
+        for (uint32_t idx = 0; idx < object_count; idx++)
+        {
+            objectIds[idx] = metaKey[idx].objectkey.key.object_id;
+        }
+
+        return bulkRemove(
+                    info->objecttype,
+                    object_count,
+                    objectIds.data(),
+                    mode,
+                    object_statuses);
+    }
+
+    switch (info->objecttype)
+    {
+        case SAI_OBJECT_TYPE_FDB_ENTRY:
+        {
+            std::vector<sai_fdb_entry_t> entries;
+            for (size_t idx = 0; idx < object_count; idx++)
+            {
+                entries.push_back(metaKey[idx].objectkey.key.fdb_entry);
+            }
+            return bulkRemove(object_count, entries.data(), mode, object_statuses);
+        }
+
+        case SAI_OBJECT_TYPE_ROUTE_ENTRY:
+        {
+            std::vector<sai_route_entry_t> entries;
+            for (size_t idx = 0; idx < object_count; idx++)
+            {
+                entries.push_back(metaKey[idx].objectkey.key.route_entry);
+            }
+            return bulkRemove(object_count, entries.data(), mode, object_statuses);
+        }
+
+        case SAI_OBJECT_TYPE_NAT_ENTRY:
+        {
+            std::vector<sai_nat_entry_t> entries;
+            for (size_t idx = 0; idx < object_count; idx++)
+            {
+                entries.push_back(metaKey[idx].objectkey.key.nat_entry);
+            }
+            return bulkRemove(object_count, entries.data(), mode, object_statuses);
+        }
+
+        default:
+            SWSS_LOG_ERROR("object type %s not implemented, FIXME", info->objecttypename);
+            return SAI_STATUS_NOT_SUPPORTED;
+    }
+}
+
+sai_status_t SaiInterface::bulkSet(
+        _Inout_ std::vector<sai_object_meta_key_t> &metaKey,
+        _In_ const sai_attribute_t **attr_list,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses)
+{
+    SWSS_LOG_ENTER();
+
+    uint32_t object_count = (uint32_t)metaKey.size();
+
+    if (!object_count)
+    {
+        SWSS_LOG_ERROR("Container with objects is empty in bulkSet");
+        return SAI_STATUS_FAILURE;
+    }
+
+    auto info = sai_metadata_get_object_type_info(metaKey[0].objecttype);
+
+    for (uint32_t idx = 0; idx < object_count; idx++)
+    {
+        if (metaKey[idx].objecttype != info->objecttype)
+        {
+            SWSS_LOG_ERROR(
+                "invalid type %s in metaKeys container with types %s",
+                metaKey[idx].objecttype,
+                info->objecttype);
+            return SAI_STATUS_FAILURE;
+        }
+    }
+
+    if (info->isobjectid)
+    {
+        std::vector<sai_object_id_t> objectIds(object_count);
+        std::vector<sai_attribute_t> attr_lists(object_count);
+
+        for (uint32_t idx = 0; idx < object_count; idx++)
+        {
+            objectIds[idx] = metaKey[idx].objectkey.key.object_id;
+            attr_lists[idx] = attr_list[idx][0];
```

Comment:
no need to do that [0] if single list would be used, you could just pass that list directy, no need to attr_lists ether